packages/search/src/elasticsearch/utils.ts

```diff
@@ -306,106 +296,65 @@ export async function detectDeathDuplicates(
-  const operatorNameLocale = user && findNameLocale(user.name)
-
-  const operatorFirstNames = operatorName?.given?.join(' ') || ''
-  const operatorFamilyName = operatorName?.family || ''
-  const operatorFirstNamesLocale = operatorNameLocale?.given?.join(' ') || ''
-  const operatorFamilyNameLocale = operatorNameLocale?.family || ''
-
-  const regLastOfficeExtension = findTaskExtension(
-    task,
-    'http://opencrvs.org/specs/extension/regLastOffice'
-  )
-
-  let office: SavedLocation | undefined
-  if (regLastOfficeExtension) {
-    office = getFromBundleById<SavedLocation>(
-      bundle,
-      regLastOfficeExtension.valueReference.reference.split('/')[1]
-    )?.resource
-  }
-
   const operationHistory = {
     operationType: body.type,
-    operatedOn: task?.lastModified,
-    rejectReason: body.rejectReason,
-    rejectComment: body.rejectComment,
-    operatorRole:
-      user?.role?.labels.find((label) => label.lang === 'en')?.label || '',
-    operatorFirstNames,
-    operatorFamilyName,
-    operatorFirstNamesLocale,
-    operatorFamilyNameLocale,
-    operatorOfficeName: office?.name || '',
-    operatorOfficeAlias: office?.alias || []
+    operatedOn: task?.lastModified
   } as IOperationHistory
 
-  if (
-    isDeclarationInStatus(body, IN_PROGRESS_STATUS) &&
-    isNotification(body) &&
-    body.eventLocationId
-  ) {
-    const facility = getFromBundleById<SavedLocation>(
-      bundle,
-      body.eventLocationId
-    ).resource
-    operationHistory.notificationFacilityName = facility?.name || ''
-    operationHistory.notificationFacilityAlias = facility?.alias || []
-  }
-
-  if (isDeclarationInStatus(body, REQUESTED_CORRECTION_STATUS)) {
-    updateOperationHistoryWithCorrection(operationHistory, task)
-  }
   body.operationHistories = body.operationHistories || []
   body.operationHistories.push(operationHistory)
 }
 
-function isDeclarationInStatus(
-  body: ICompositionBody,
-  status: string
-): boolean {
-  return (body.type && body.type === status) || false
-}
+export const assign = (body: SearchDocument, bundle: SavedBundle) => {
+  const task = getTaskFromSavedBundle(bundle)
 
-function isNotification(body: ICompositionBody): boolean {
-  return (
-    (body.compositionType &&
-      NOTIFICATION_TYPES.includes(body.compositionType)) ||
-    false
+  const assignedExtension = findExtension(
+    ASSIGNED_EXTENSION_URL,
+    task.extension
   )
+  const regLastUserExtension = findExtension(
+    LAST_USER_EXTENSION_URL,
+    task.extension
+  )
+  const regLastOffice = findLastOfficeFromSavedBundle(bundle)
+  if (assignedExtension) {
+    const userId = resourceIdentifierToUUID(
+      regLastUserExtension!.valueReference.reference
+    )
+    const user = getPractitioner(userId, bundle)
+
+    const userName = user && findName(NAME_EN, user.name)
+    const userFirstNames = userName?.given?.join(' ') || ''
+    const userFamilyName = userName?.family || ''
+
+    body.assignment = {
+      userId,
```

Comment:
True.

https://github.com/opencrvs/opencrvs-core/blob/b26d30313ca5b956c362ce5ed642d52e7cd09574/packages/client/src/components/interface/DownloadButton.tsx#L353
for instance here we check if the userId's match. Wonder if we should rather resolve the userId in gateway resolvers from practitionerId to the Mongo userId to keep most of the functionality similar to previous. Maybe ElasticSearch could store "practitionerId" instead of "userId" and then change it around in gateway. Downside is a new API call.